contracts/gateway/GatewayRedeemInterface.sol

```diff
@@ -1,55 +1,106 @@
 pragma solidity ^0.5.0;
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+// Auxiliary Chain: CoGateway Contract
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
 
+/**
+ *  @title GatewayRedeemInterface interface.
+ *
+ *  @notice An interface to call CoGateway redeem functions.
+ */
 contract GatewayRedeemInterface {
 
-    function isActivated() external returns (bool);
-    function bounty() external returns (uint256);
-
-    function redeem(
-        uint256 _amount,
-        address _beneficiary,
-        address _facilitator,
-        uint256 _gasPrice,
-        uint256 _nonce,
-        bytes32 _hashLock
-    )
-        public
-        payable
-        returns (bytes32 messageHash_);
-
-    function processRedemption(
-        bytes32 _messageHash,
-        bytes32 _unlockSecret
-    )
-        external
-        returns (uint256 redeemAmount);
-
-    function processRedemptionWithProof(
-        bytes32 _messageHash,
-        bytes calldata _rlpEncodedParentNodes,
-        uint256 _blockHeight,
-        uint256 _messageStatus
-    )
-        external
-        returns (uint256 redeemAmount);
-
-    function revertRedemption(
-        bytes32 _messageHash
-    )
-        external
-        returns (
-            address redeemer_,
-            bytes32 intentHash_,
-            uint256 nonce_,
-            uint256 gasPrice_
-        );
-
-    function processRevertRedemption(
```

Comment:
is there no equivalent to `processRevertRedemption` anymore? What about `isActivated()`, `redeem`, etc.? Why are those removed as part of this PR?

Reply:
The interface was outdated. `processRevertRedemption` and `isActivated` were not needed.  
`redeem` and `RevertRedeem` are available. It has all the methods that may be required to redeem process